experanto/interpolators.py

```diff
@@ -478,6 +479,64 @@ def rescale_frame(self, frame: np.array) -> np.array:
         )
 
 
+class TimeIntervalInterpolator(Interpolator):
+    def __init__(self, root_folder: str, cache_data: bool = False, **kwargs):
+        super().__init__(root_folder)
+        self.cache_data = cache_data
+
+        meta = self.load_meta()
+        self.labels = list(sorted(meta["labels"].keys()))
+        self.start_time = meta["start_time"]
+        self.end_time = meta["end_time"]
+        self.valid_interval = TimeInterval(self.start_time, self.end_time)
+
+        if self.cache_data:
+            self.label_data = {
```

Comment:
opinionated nitpick (e.g. feel free to ignore) but I would call it 
```suggestion
            self.labeled_interval = {
```
as they are intervals while data could anything